app/src/main/java/com/keylesspalace/tusky/fragment/ViewThreadFragment.java

```diff
@@ -621,6 +639,28 @@ private void handleReblogEvent(ReblogEvent event) {
         adapter.setItem(posAndStatus.first, newViewData, true);
     }
 
+    private void handleBookmarkEvent(BookmarkEvent event) {
+        Pair<Integer, Status> posAndStatus = findStatusAndPos(event.getStatusId());
+        if (posAndStatus == null) return;
+
+        boolean bookmark = event.getBookmark();
+        posAndStatus.second.setBookmarked(bookmark);
```

Comment:
Is it okay that both status and reblog are bookmarked? Should one of them be bookmarked?

Reply:
We do it like this for favourites since ever and I don't want to debug why exactly :D